src/QsCompiler/TextProcessor/QsExpressionParsing.fs

```diff
@@ -183,9 +183,41 @@ let internal numericLiteral =
             let trimmed = nl.String.TrimStart '+'
             if format = 10 || format = 0 then trimmed else trimmed.Substring 2 |> sprintf "0%s" // leading 0 is required to keep numbers positive
         let isInt    = nl.IsInteger     && format <> 0                  && nl.SuffixLength = 0 // any format is fine here
-        let isBigInt = nl.IsInteger     && (format = 10 || format = 16) && nl.SuffixLength = 1 && System.Char.ToUpperInvariant(nl.SuffixChar1) = 'L'
+        let isBigInt = nl.IsInteger     && format <> 0  && nl.SuffixLength = 1 && System.Char.ToUpperInvariant(nl.SuffixChar1) = 'L'
         let isDouble = not nl.IsInteger && format = 10                  && nl.SuffixLength = 0 
         let returnWithRange kind = preturn (kind, range)
+        let binaryStringToHex binaryString:string = 
+            // first pad 0's so that length is multiple of 4, so we can match from left rather than right
+            let nZeroPad = (4 - string(binaryString).Length % 4) % 4 // if str.Length is already multiple of 4 then we don't pad
+            let rec padZero(curString, numPad) =
+                match numPad with
+                | 0 -> (curString,0)
+                | _ -> padZero(curString |>sprintf("0%s"), numPad-1)
+            let (binaryString, _) = padZero(binaryString, nZeroPad)
+            // now match from left
+            let biList = binaryString |> Seq.toList
+            let rec binaryListToHex binaryList =
+                match binaryList with
+                | [] -> []
+                | '0'::'0'::'0'::'0'::rest -> '0'::binaryListToHex rest
+                | '0'::'0'::'0'::'1'::rest -> '1'::binaryListToHex rest
+                | '0'::'0'::'1'::'0'::rest -> '2'::binaryListToHex rest
+                | '0'::'0'::'1'::'1'::rest -> '3'::binaryListToHex rest
+                | '0'::'1'::'0'::'0'::rest -> '4'::binaryListToHex rest
+                | '0'::'1'::'0'::'1'::rest -> '5'::binaryListToHex rest
+                | '0'::'1'::'1'::'0'::rest -> '6'::binaryListToHex rest
+                | '0'::'1'::'1'::'1'::rest -> '7'::binaryListToHex rest
+                | '1'::'0'::'0'::'0'::rest -> '8'::binaryListToHex rest
+                | '1'::'0'::'0'::'1'::rest -> '9'::binaryListToHex rest
+                | '1'::'0'::'1'::'0'::rest -> 'A'::binaryListToHex rest
+                | '1'::'0'::'1'::'1'::rest -> 'B'::binaryListToHex rest
+                | '1'::'1'::'0'::'0'::rest -> 'C'::binaryListToHex rest
+                | '1'::'1'::'0'::'1'::rest -> 'D'::binaryListToHex rest
+                | '1'::'1'::'1'::'0'::rest -> 'E'::binaryListToHex rest
+                | '1'::'1'::'1'::'1'::rest -> 'F'::binaryListToHex rest
```

Comment:
`Seq.chunkBySize 4` might come in handy here - I think something like `binaryString |> Seq.chunkBySize 4 |> Seq.map System.String` will give you a sequence of strings of length 4, so you could map over them with

```fsharp
// ...
| "1001" -> "9"
| "1010" -> "A"
// ...
```

At that point, you might be able to avoid enumerating all the cases by round-tripping from string-to-int-to-string like:

```fsharp
> System.Convert.ToByte("1010", 2).ToString "X";;
val it : string = "A"
```

Reply:
latest PR no longer enumerates - but still need to enumerate octal->binary since "X" is the only keyword that's recognized

Reply:
Hmm, it looks like `System.Convert.ToString (x, 2)` will convert a number to a binary string. I wonder if there is a way to go directly from octal to hex, but the digits don't seem to line up nicely, so it's probably easier to go through binary.

Reply:
Oh, I think you can convert 4 digits of octal to 3 digits of hex. Since you can use `System.Convert` for both ways instead of enumerating the cases, that might be better?

Reply:
I tried that first:) -  it doesn't work, since Byte only allows for values up to 255, and 4 digits of octal is much bigger than that

Reply:
You can use `System.Convert.ToInt32` (or `ToInt16`, or `ToInt64`, also):

```fsharp
> System.Convert.ToInt32 ("7777", 8);;
val it : int = 4095
```

Reply:
Duh... never used F# before haha. Updated to this in latest PR.